<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CQDlgJointDyn</class>
 <widget class="QDialog" name="CQDlgJointDyn">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>430</width>
    <height>243</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Joint Dynamic Properties</string>
  </property>
  <property name="locale">
   <locale language="English" country="UnitedStates"/>
  </property>
  <widget class="QToolButton" name="qqApplyParams">
   <property name="geometry">
    <rect>
     <x>260</x>
     <y>210</y>
     <width>161</width>
     <height>18</height>
    </rect>
   </property>
   <property name="palette">
    <palette>
     <active>
      <colorrole role="WindowText">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>0</red>
         <green>0</green>
         <blue>0</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="Button">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>220</red>
         <green>220</green>
         <blue>220</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="Light">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>255</red>
         <green>255</green>
         <blue>255</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="Midlight">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>237</red>
         <green>237</green>
         <blue>237</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="Dark">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>110</red>
         <green>110</green>
         <blue>110</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="Mid">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>147</red>
         <green>147</green>
         <blue>147</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="Text">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>0</red>
         <green>0</green>
         <blue>0</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="BrightText">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>255</red>
         <green>255</green>
         <blue>255</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="ButtonText">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>0</red>
         <green>0</green>
         <blue>0</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="Base">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>255</red>
         <green>255</green>
         <blue>255</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="Window">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>220</red>
         <green>220</green>
         <blue>220</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="Shadow">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>0</red>
         <green>0</green>
         <blue>0</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="AlternateBase">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>237</red>
         <green>237</green>
         <blue>237</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="ToolTipBase">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>255</red>
         <green>255</green>
         <blue>220</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="ToolTipText">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>0</red>
         <green>0</green>
         <blue>0</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="PlaceholderText">
       <brush brushstyle="NoBrush">
        <color alpha="128">
         <red>0</red>
         <green>0</green>
         <blue>0</blue>
        </color>
       </brush>
      </colorrole>
     </active>
     <inactive>
      <colorrole role="WindowText">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>0</red>
         <green>0</green>
         <blue>0</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="Button">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>220</red>
         <green>220</green>
         <blue>220</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="Light">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>255</red>
         <green>255</green>
         <blue>255</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="Midlight">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>237</red>
         <green>237</green>
         <blue>237</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="Dark">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>110</red>
         <green>110</green>
         <blue>110</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="Mid">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>147</red>
         <green>147</green>
         <blue>147</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="Text">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>0</red>
         <green>0</green>
         <blue>0</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="BrightText">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>255</red>
         <green>255</green>
         <blue>255</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="ButtonText">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>0</red>
         <green>0</green>
         <blue>0</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="Base">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>255</red>
         <green>255</green>
         <blue>255</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="Window">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>220</red>
         <green>220</green>
         <blue>220</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="Shadow">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>0</red>
         <green>0</green>
         <blue>0</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="AlternateBase">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>237</red>
         <green>237</green>
         <blue>237</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="ToolTipBase">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>255</red>
         <green>255</green>
         <blue>220</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="ToolTipText">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>0</red>
         <green>0</green>
         <blue>0</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="PlaceholderText">
       <brush brushstyle="NoBrush">
        <color alpha="128">
         <red>0</red>
         <green>0</green>
         <blue>0</blue>
        </color>
       </brush>
      </colorrole>
     </inactive>
     <disabled>
      <colorrole role="WindowText">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>110</red>
         <green>110</green>
         <blue>110</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="Button">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>220</red>
         <green>220</green>
         <blue>220</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="Light">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>255</red>
         <green>255</green>
         <blue>255</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="Midlight">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>237</red>
         <green>237</green>
         <blue>237</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="Dark">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>110</red>
         <green>110</green>
         <blue>110</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="Mid">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>147</red>
         <green>147</green>
         <blue>147</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="Text">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>110</red>
         <green>110</green>
         <blue>110</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="BrightText">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>255</red>
         <green>255</green>
         <blue>255</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="ButtonText">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>110</red>
         <green>110</green>
         <blue>110</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="Base">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>220</red>
         <green>220</green>
         <blue>220</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="Window">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>220</red>
         <green>220</green>
         <blue>220</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="Shadow">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>0</red>
         <green>0</green>
         <blue>0</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="AlternateBase">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>220</red>
         <green>220</green>
         <blue>220</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="ToolTipBase">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>255</red>
         <green>255</green>
         <blue>220</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="ToolTipText">
       <brush brushstyle="SolidPattern">
        <color alpha="255">
         <red>0</red>
         <green>0</green>
         <blue>0</blue>
        </color>
       </brush>
      </colorrole>
      <colorrole role="PlaceholderText">
       <brush brushstyle="NoBrush">
        <color alpha="128">
         <red>0</red>
         <green>0</green>
         <blue>0</blue>
        </color>
       </brush>
      </colorrole>
     </disabled>
    </palette>
   </property>
   <property name="focusPolicy">
    <enum>Qt::StrongFocus</enum>
   </property>
   <property name="text">
    <string>Apply to selection</string>
   </property>
  </widget>
  <widget class="QToolButton" name="qqAdjustEngineProperties">
   <property name="geometry">
    <rect>
     <x>260</x>
     <y>180</y>
     <width>161</width>
     <height>18</height>
    </rect>
   </property>
   <property name="focusPolicy">
    <enum>Qt::StrongFocus</enum>
   </property>
   <property name="text">
    <string>Engine properties</string>
   </property>
  </widget>
  <widget class="QStackedWidget" name="qqStack">
   <property name="geometry">
    <rect>
     <x>-1</x>
     <y>40</y>
     <width>421</width>
     <height>131</height>
    </rect>
   </property>
   <property name="currentIndex">
    <number>3</number>
   </property>
   <widget class="QWidget" name="page">
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>0</y>
       <width>391</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>The joint is not controlled</string>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="page_2">
    <widget class="QLabel" name="qqForceMode_forceLabel">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>0</y>
       <width>301</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Constant torque</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="qqForceMode_force">
     <property name="geometry">
      <rect>
       <x>340</x>
       <y>0</y>
       <width>81</width>
       <height>18</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="page_3">
    <widget class="QLineEdit" name="qqVelocityMode_velocity">
     <property name="geometry">
      <rect>
       <x>340</x>
       <y>0</y>
       <width>81</width>
       <height>18</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="qqVelocityMode_force">
     <property name="geometry">
      <rect>
       <x>340</x>
       <y>20</y>
       <width>81</width>
       <height>18</height>
      </rect>
     </property>
    </widget>
    <widget class="QCheckBox" name="qqVelocityMode_motorLock">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>100</y>
       <width>291</width>
       <height>18</height>
      </rect>
     </property>
     <property name="focusPolicy">
      <enum>Qt::StrongFocus</enum>
     </property>
     <property name="text">
      <string>Lock motor when target velocity is zero</string>
     </property>
    </widget>
    <widget class="QLabel" name="qqVelocityMode_velocityLabel">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>0</y>
       <width>261</width>
       <height>18</height>
      </rect>
     </property>
     <property name="text">
      <string>Target velocity [xxx]</string>
     </property>
    </widget>
    <widget class="QLabel" name="qqVelocityMode_forceLabel">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>281</width>
       <height>18</height>
      </rect>
     </property>
     <property name="text">
      <string>Max. force/torque</string>
     </property>
    </widget>
    <widget class="QLabel" name="qqVelocityMode_maxAccelLabel">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>60</y>
       <width>191</width>
       <height>18</height>
      </rect>
     </property>
     <property name="text">
      <string>Max. acceleration</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="qqVelocityMode_maxJerk">
     <property name="geometry">
      <rect>
       <x>340</x>
       <y>80</y>
       <width>81</width>
       <height>18</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="qqVelocityMode_maxJerkLabel">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>80</y>
       <width>191</width>
       <height>18</height>
      </rect>
     </property>
     <property name="text">
      <string>Max. jerk</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="qqVelocityMode_maxAccel">
     <property name="geometry">
      <rect>
       <x>340</x>
       <y>60</y>
       <width>81</width>
       <height>18</height>
      </rect>
     </property>
    </widget>
    <widget class="QCheckBox" name="qqVelocityMode_ruckig">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>211</width>
       <height>18</height>
      </rect>
     </property>
     <property name="focusPolicy">
      <enum>Qt::StrongFocus</enum>
     </property>
     <property name="text">
      <string>Motion profile control</string>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="page_4">
    <widget class="QLabel" name="qqPositionMode_forceLabel">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>281</width>
       <height>18</height>
      </rect>
     </property>
     <property name="text">
      <string>Max. force/torque</string>
     </property>
    </widget>
    <widget class="QLabel" name="qqPositionMode_positionLabel">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>0</y>
       <width>271</width>
       <height>18</height>
      </rect>
     </property>
     <property name="text">
      <string>Target position [xxx]</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="qqPositionMode_cb">
     <property name="geometry">
      <rect>
       <x>340</x>
       <y>60</y>
       <width>71</width>
       <height>18</height>
      </rect>
     </property>
     <property name="focusPolicy">
      <enum>Qt::StrongFocus</enum>
     </property>
     <property name="text">
      <string>CB</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="qqPositionMode_force">
     <property name="geometry">
      <rect>
       <x>340</x>
       <y>20</y>
       <width>81</width>
       <height>18</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="qqPositionMode_position">
     <property name="geometry">
      <rect>
       <x>340</x>
       <y>0</y>
       <width>81</width>
       <height>18</height>
      </rect>
     </property>
    </widget>
    <widget class="QCheckBox" name="qqPositionMode_ruckig">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>60</y>
       <width>241</width>
       <height>19</height>
      </rect>
     </property>
     <property name="text">
      <string>Motion profile control</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <property name="checked">
      <bool>false</bool>
     </property>
     <property name="autoExclusive">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QLineEdit" name="qqPositionMode_maxJerk">
     <property name="geometry">
      <rect>
       <x>340</x>
       <y>100</y>
       <width>81</width>
       <height>18</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="qqPositionMode_maxVelLabel">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>191</width>
       <height>18</height>
      </rect>
     </property>
     <property name="text">
      <string>Max. velocity</string>
     </property>
    </widget>
    <widget class="QLabel" name="qqPositionMode_maxAccelLabel">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>80</y>
       <width>191</width>
       <height>18</height>
      </rect>
     </property>
     <property name="text">
      <string>Max. acceleration</string>
     </property>
    </widget>
    <widget class="QLabel" name="qqPositionMode_maxJerkLabel">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>100</y>
       <width>191</width>
       <height>18</height>
      </rect>
     </property>
     <property name="text">
      <string>Max. jerk</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="qqPositionMode_maxVel">
     <property name="geometry">
      <rect>
       <x>340</x>
       <y>40</y>
       <width>81</width>
       <height>18</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="qqPositionMode_maxAccel">
     <property name="geometry">
      <rect>
       <x>340</x>
       <y>80</y>
       <width>81</width>
       <height>18</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="page_5">
    <widget class="QCheckBox" name="qqSpringMode_cb">
     <property name="geometry">
      <rect>
       <x>340</x>
       <y>60</y>
       <width>41</width>
       <height>18</height>
      </rect>
     </property>
     <property name="focusPolicy">
      <enum>Qt::StrongFocus</enum>
     </property>
     <property name="text">
      <string>CB</string>
     </property>
    </widget>
    <widget class="QLabel" name="qqSpringMode_positionLabel">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>0</y>
       <width>271</width>
       <height>18</height>
      </rect>
     </property>
     <property name="text">
      <string>Target position [xxx]</string>
     </property>
    </widget>
    <widget class="QLabel" name="qqKLabel">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>221</width>
       <height>18</height>
      </rect>
     </property>
     <property name="text">
      <string>Spring constant K</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="qqK">
     <property name="geometry">
      <rect>
       <x>340</x>
       <y>20</y>
       <width>81</width>
       <height>18</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="qqCLabel">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>211</width>
       <height>18</height>
      </rect>
     </property>
     <property name="text">
      <string>Spring constant C</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="qqC">
     <property name="geometry">
      <rect>
       <x>340</x>
       <y>40</y>
       <width>81</width>
       <height>18</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="qqSpringMode_position">
     <property name="geometry">
      <rect>
       <x>340</x>
       <y>0</y>
       <width>81</width>
       <height>18</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="page_6">
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>0</y>
       <width>391</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>A custom controller can be written in the joint callback function</string>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QComboBox" name="qqCombo">
   <property name="geometry">
    <rect>
     <x>310</x>
     <y>10</y>
     <width>111</width>
     <height>18</height>
    </rect>
   </property>
  </widget>
  <widget class="QLabel" name="qqModeLabel">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>10</y>
     <width>271</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>Control mode</string>
   </property>
  </widget>
 </widget>
 <tabstops>
  <tabstop>qqCombo</tabstop>
  <tabstop>qqForceMode_force</tabstop>
  <tabstop>qqVelocityMode_velocity</tabstop>
  <tabstop>qqVelocityMode_force</tabstop>
  <tabstop>qqVelocityMode_motorLock</tabstop>
  <tabstop>qqPositionMode_position</tabstop>
  <tabstop>qqPositionMode_force</tabstop>
  <tabstop>qqPositionMode_cb</tabstop>
  <tabstop>qqPositionMode_ruckig</tabstop>
  <tabstop>qqPositionMode_maxVel</tabstop>
  <tabstop>qqPositionMode_maxAccel</tabstop>
  <tabstop>qqPositionMode_maxJerk</tabstop>
  <tabstop>qqSpringMode_position</tabstop>
  <tabstop>qqK</tabstop>
  <tabstop>qqSpringMode_cb</tabstop>
  <tabstop>qqC</tabstop>
  <tabstop>qqAdjustEngineProperties</tabstop>
  <tabstop>qqApplyParams</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
